src/Chapter3.hs

```diff
@@ -373,6 +377,29 @@ after the fight. The battle has the following possible outcomes:
    doesn't earn any money and keeps what they had before.
 
 -}
+data Knight = MkKnight
+  { knightName :: String
+  , knightHealth :: Int
+  , knightAttack :: Int
+  , knightGold :: Int} deriving (Show)
+  
+data Monster = MkMonster
+  { monsterName :: String
+  , monsterHealth :: Int
+  , monsterAttack :: Int
+  , monsterGold :: Int} deriving (Show)
+  
+fight :: Knight -> Monster -> Int
+fight knight monster =
+  let
+    newMonsterHealth = monsterHealth monster - knightAttack knight
+    newKnightHealth = knightHealth knight - monsterAttack monster
+  in
+    if newMonsterHealth <= 0 then 
+      knightGold knight + monsterGold monster
+    else
+      if newKnightHealth <= 0 then -1 
+      else knightGold knight
```

Comment:
Usually, nested `if` can be replaced with guards, in this case, you can:
```haskell
    | newMonsterHealth <= 0 = knightGold knight + monsterGold monster
    | newKnightHealth <= 0 = -1 
    | otherwise = knightGold knight
```

(Note, that in this case, you would need to use `where` instead of `let` to have it in scope in each of the guards casing 🙂